autometa/common/external/samtools.py

```diff
@@ -47,32 +47,30 @@ def sort(sam, bam, nproc=mp.cpu_count()):
     Raises
     ------
     TypeError
-        nproc must be an interger greater than zero
+        nproc must be an integer greater than zero
     ChildProcessError
         Samtools did not run successfully, returns the return code from subprocessing call
     """
 
-    if type(nproc) is not int or nproc == 0:
+    if type(nproc) is not int or nproc <= 0:
         raise TypeError(f'nproc must be an integer greater than zero! Given: {nproc}')
     samtools_out_dir = os.path.dirname(os.path.abspath(bam))
-    outfile = os.path.basename(bam)
-    tempdir = tempfile.mkdtemp(suffix = None, prefix = 'samtools', dir = samtools_out_dir)
-    samtools_outfile = os.path.join(samtools_out_dir, outfile)
-    temp_outfile = os.path.join(tempdir, outfile)
-    cmd = f'samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {temp_outfile}'
-    logger.debug(f'cmd: {cmd}')
     samtools_err = os.path.join(samtools_out_dir, 'samtools.err')
     samtools_out = os.path.join(samtools_out_dir, 'samtools.out')
-    with open(samtools_out, 'w') as stdout, open(samtools_err, 'w') as stderr:
-        retcode = subprocess.call(cmd, stdout=stdout, stderr=stderr, shell=True)
-        if retcode != 0:
-            shutil.rmtree(tempdir, ignore_errors=True)
-            raise ChildProcessError(f'Sort failed, retcode: {retcode}')
-        else:
-            shutil.move(temp_outfile, samtools_outfile)
-            shutil.rmtree(tempdir, ignore_errors=True)
-    os.remove(samtools_err)
-    os.remove(samtools_out)
+    outfile = os.path.basename(bam)
+    samtools_outfile = os.path.join(samtools_out_dir, outfile)
+    with tempfile.TemporaryDirectory(suffix = None, prefix = 'samtools', dir = samtools_out_dir) as tempdir:    
```

Comment:
A few things here

1. When passing arguments into a function, recommended style is to  place args with their keyword. I.e. `suffix=None, prefix='samtools', dir='samtools_out_dir'`
2. Since you are passing `None` into suffix, you can omit this.
3. Do any of these arguments need to be supplied? Do we care where the tmpdir is located and do we care what it is named? If no, we can omit all of these arguments and this should allow you to be more concise, allowing for better readability.

I have been following this styling guide: [google style guide - whitespace](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#36-whitespace)

Reply:
I don't think we care where it is located and what it is named. I looked at some of the scripts where you have made a ``temp`` directory and since you had supplied these arguments as well I thought it is a good practice to supply them.